packages/AI/VectorDupe/src/duplicateRecordDetector.ts

```diff
@@ -313,14 +325,14 @@ export class DuplicateRecordDetector extends VectorBase {
         return list;
     }
 
-    private async createListDetailsForDupeRun(recordIDs: PrimaryKeyValueBase[], listID: number): Promise<void> {
+    private async createListDetailsForDupeRun(recordIDs: CompositeKey[], listID: number): Promise<void> {
         const md: Metadata = new Metadata();
 
         for(const recordID of recordIDs){
             const listDetail: ListDetailEntity = await md.GetEntityObject<ListDetailEntity>('List Details');
             listDetail.NewRecord();
             listDetail.ListID = listID;
-            listDetail.RecordID = recordID.GetCompositeKeyAsSQLString();
+            listDetail.RecordID = recordID.ToString();
```

Comment:
@JS-BC  this will be very inefficient as written. What is the reason for having a list of RecordIDs - the INPUT into this entire process should be a Source ListID, I don't know why this stuff is still here?

Reply:
Also @JS-BC  if you ever needed to bulk create records in the DB like this, you should use a batching approach and `Promise.All` to group batches of records to the DB, like 10 or 20 at a time and run those in parallel. Ask ChatGPT for help doing that, but here I don't know why this is still here at all